### PR TITLE
Create new loader if view projection changes

### LINKED ISSUE #15161: Image WMS does not consider projection after view changed
**Describe the bug**
After view changed (with a new projection), an ImageWMS source generate request with the old projection

**To Reproduce**
Instantiate a map, with a view with EPSG:4326 projection.
Add a Image layer with a ImageWMS source to the map
Instantiate a new view with EPSG:3857 projection
Zoom/pan on the map, and check the layer's request

**Expected behavior**
the layer has to generate a request with the EPSG:3857 projection

This is a regression of the v8.0.0

### PR DESCRIPTION
**Pull request by mike-000**
Fixes #15161

### CODE REVIEW

File: src/ol/source/ImageArcGISRest.js [hunk 1]
```diff
@@ -141,8 +147,9 @@ class ImageArcGISRest extends ImageSource {
     if (this.url_ === undefined) {
       return null;
     }
-    if (!this.loader) {
+    if (!this.loader || this.loaderProjection_ !== projection) {
```
Comment (ahocevar) [inline src/ol/source/ImageArcGISRest.js:150]: The `equivalent()` function should be used here

File: src/ol/source/ImageMapGuide.js [hunk 2]
```diff
@@ -141,8 +147,9 @@ class ImageMapGuide extends ImageSource {
     if (this.url_ === undefined) {
       return null;
     }
-    if (!this.loader) {
+    if (!this.loader || this.loaderProjection_ !== projection) {
```
Comment (ahocevar) [inline src/ol/source/ImageMapGuide.js:150]: Same

File: src/ol/source/ImageWMS.js [hunk 3]
```diff
@@ -195,8 +201,9 @@ class ImageWMS extends ImageSource {
     if (this.url_ === undefined) {
       return null;
     }
-    if (!this.loader) {
+    if (!this.loader || this.loaderProjection_ !== projection) {
```
Comment (ahocevar) [inline src/ol/source/ImageWMS.js:204]: Same

**Review by ahocevar — CHANGES_REQUESTED** [review]

File: src/ol/source/ImageArcGISRest.js (hunk 1 above)
Comment (mike-000) [reply src/ol/source/ImageArcGISRest.js:150]: The code replicates the behavior before version 8 - if the view projection was EPSG:3857 the image was requested for CRS EPSG:3857, if the view projection was changed to EPSG:900913 the image was requested for CRS EPSG:900913.  Ditto for EPSG:4326/CRS:84, etc.

**Review by ahocevar — APPROVED** [review]

**Comment by ahocevar** [comment]
Obviously a test would be good to avoid future regressions on this.

**Review by ahocevar — APPROVED** [review]

**Comment by ahocevar** [comment]
Thanks, @mike-000

**Comment by KyllianGautier** [comment]
Thanks you very much guys ❤️

**Pull request merged**